### DISCUSSION TITLE
Adding Software

### DISCUSSION THREAD

**Discussion by Kaympe20** [community/community#59368] [Codespaces] [answered]
### Select Topic Area

Question

### Body

How do I add compiled binaries that are part of workflow to my Codespaces instance?

**Comment by Fengyi-Li** [comment, marked as answer]
Yes, you can do with different options.
1. Commit your binaries to your repository and use it in Codespaces.
2. Add commands to dev container configuration to generate binaries you want, then it will be created in Codespaces container. [Ref](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/introduction-to-dev-containers)
3. Codespaces has a prebuild feature https://docs.github.com/en/codespaces/prebuilding-your-codespaces/configuring-prebuilds#configuring-prebuilds that you can use to build your binaries in template and then you can use it in Codespace instances without baking time.
